platform/linuxbsd/detect.py

```diff
@@ -111,6 +110,9 @@ def configure(env):
             env["CC"] = "clang"
             env["CXX"] = "clang++"
         env.extra_suffix = ".llvm" + env.extra_suffix
+    else:
+        env["RANLIB"] = "gcc-ranlib"
+        env["AR"] = "gcc-ar"
```

Comment:
Why force GCC stuff here? I see that this was moved, but now GCC stuff will be always forced on instead of using the system utilities, even without LTO.

Reply:
I just moved it, it seemed weird to force this for LTO and not for non-LTO. I can move it back but this hacky either way, requires more work.

Reply:
I see why you did it, but I think that the idea was to force GCC's `ranlib` and `ar` just when LTOing since that's required by GCC for it.

From the gcc man page:
```
           If object files containing GIMPLE bytecode are stored in a library
           archive, say libfoo.a, it is possible to extract and use them in an
           LTO link if you are using a linker with plugin support.  To create
           static libraries suitable for LTO, use gcc-ar and gcc-ranlib
           instead of ar and ranlib; to show the symbols of object files with
           GIMPLE bytecode, use gcc-nm.  Those commands require that ar,
           ranlib and nm have been compiled with plugin support.  At link
           time, use the flag -fuse-linker-plugin to ensure that the library
           participates in the LTO optimization process:

                   gcc -o myprog -O2 -flto -fuse-linker-plugin a.o b.o -lfoo
```

Reply:
Thanks for the details, I'll move it back :+1: